src/main/kotlin/com/anaplan/engineering/vdmgradleplugin/TypeChecker.kt

```diff
@@ -50,25 +51,41 @@ open class VdmTypeCheckTask(private val includeTests: Boolean) : OvertureTask()
         @InputFiles
         get() = project.locateAllSpecifications(dialect, includeTests)
 
-    val generatedLibFile: File
+    // Only needed to ensure caching of the Gradle task
+    val statusFile: File
         @OutputFile
-        get() = project.generatedLibFile
-
+        get() = File(project.vdmBuildDir, "typeCheckStatus.log")
 
     override fun exec() {
         logger.info("VDM dialect: $dialect")
-        jvmArgs = project.vdmConfig.overtureJvmArgs
-        super.setArgs(constructArgs())
-        super.setClasspath(project.files(createClassPathJar()))
-        super.exec()
+        if (specificationFiles.isEmpty) {
+            logger.info("No files found")
+        } else {
+            jvmArgs = project.vdmConfig.overtureJvmArgs
+            super.setArgs(constructArgs())
+            super.setClasspath(project.files(createClassPathJar()))
+            super.exec()
+            writeStatus()
+        }
+    }
+
+    private fun writeStatus() {
+        val exitStatus = super.getExecutionResult().get().exitValue
```

Comment:
It's a bit odd to explicitly call `super` here -- if you had overridden in this class you would want that one to be used 

standard practice is to only use `super` when you override a method and want to call the parent method as well as your extended logic

Reply:
as per L67 infact :) 

don't know why they are there on L65/66 -- can we get rid of those too

Reply:
> don't know why they are there on L65/66 -- can we get rid of those too

Indeed, that's why I called `super` here.